### ISSUE TITLE
GHC panic: Bogus proof for stuck type family.

### ISSUE THREAD

**Issue by kephas** [polysemy-research/polysemy#463]
In this code:

https://github.com/kephas/xdg-basedir-compliant/tree/polysemy-bug

Running `cabal build` triggers this error:

```
ghc: panic! (the 'impossible' happened)
  (GHC version 9.0.2:
        Bogus proof for stuck type family.

This means there's a type error in your program, but the fact that
you're seeing this message is a bug in `polysemy-plugin`.

Please file a bug at https://github.com/polysemy-research/polysemy
with a minimal reproduction for how you managed to get this error.

CallStack (from HasCallStack):
  error, called at src/Polysemy/Plugin/Fundep.hs:288:9 in polysemy-plugin-0.4.1.1-EpBKoXMm5HlTiTBA0k24:Polysemy.Plugin.Fundep

Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug

cabal: Failed to build xdg-basedir-compliant-1.0.0 (which is required by
test:xdg-test from xdg-basedir-compliant-1.0.0).
```

**Comment by kephas** [comment]
Commit [75bf5fd1a](https://github.com/kephas/xdg-basedir-compliant/commit/75bf5fd1afaebd9d1b70c15368719cd6a162470a) contains the code that triggers the error.

**Comment by googleson78** [comment]
Can you try with a newer polysemy and polysemy-plugin? It appears `0.4.3.1` is available on `unstable` and `0.4.3.0` is available on 22.05.

Also, just to be sure, what's your polysemy version?

**Comment by kephas** [comment]
I updated my system in the meantime and, indeed, it's fixed, kudos!
